packages/core/src/bags/redux/middlewares/getOperationsOnBagRequestSuccess.js

```diff
@@ -0,0 +1,54 @@
+import {
+  ADD_ITEM_TO_BAG_SUCCESS,
+  DELETE_BAG_ITEM_SUCCESS,
+  GET_BAG_SUCCESS,
+  UPDATE_BAG_ITEM_SUCCESS,
+} from '../actionTypes';
+import {
+  doGetBagOperation,
+  resetBagOperationsEntities,
+  resetState,
+} from '../actions';
+import { getBag, getBagId } from '../selectors';
+import { getBagOperation } from '../../client';
+
+/**
+ * Middleware to fetch bag operation if a bag request is successful.
+ *
+ * @function getOperationsOnBagRequestSuccess
+ * @memberof module:bags/middlewares
+ *
+ * @param {object} store - Redux store at the moment.
+ *
+ * @returns {Function} Redux middleware.
+ */
+export default store => next => action => {
+  if (
+    action.type === ADD_ITEM_TO_BAG_SUCCESS ||
+    action.type === GET_BAG_SUCCESS ||
+    action.type === UPDATE_BAG_ITEM_SUCCESS ||
+    action.type === DELETE_BAG_ITEM_SUCCESS
+  ) {
+    const bagId = getBagId(store.getState());
+    const controls = getBag(store.getState())?.['@controls'];
+
+    if (!controls) {
+      return next(action);
+    }
+
+    store.dispatch(resetState(['bagOperations']));
+    store.dispatch(resetBagOperationsEntities());
+
+    const operationIds = Object.values(controls)
+      .map(({ href }) => href?.split('/')?.pop())
+      .filter(Boolean);
+
+    operationIds.forEach(operationId => {
+      const getOperation = doGetBagOperation(getBagOperation);
```

Comment:
The action factory can be set before the forEach as it does not depend on it.

Reply:
Done!